### PR TITLE
support dmg notarizing

### PR DESCRIPTION
**Pull request by shiqi**
support notarizing dmg files

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -31,31 +31,17 @@ export type NotarizeOptions = NotarizeStartOptions;
...
-      {
-        cwd: path.dirname(opts.appPath),
-      },
-    );
-    if (zipResult.code !== 0) {
-      throw new Error(`Failed to zip application, exited with code: ${zipResult.code}\n\n${zipResult.output}`);
-    }
-    d('zip succeeded, attempting to upload to apple');
-
+  //zip the app unless it is dmg
```
Comment (malept) [inline src/index.ts]: This isn't a very useful comment.
```suggestion
```

File: src/index.ts [hunk 2]
```diff
@@ -31,31 +31,17 @@ export type NotarizeOptions = NotarizeStartOptions;
 
 export async function startNotarize(opts: NotarizeStartOptions): Promise<NotarizeResult> {
   d('starting notarize process for app:', opts.appPath);
+  var appPath = opts.appPath;
```
Comment (malept) [inline src/index.ts]:
```suggestion
  let uploadPath = opts.appPath;
```

File: src/index.ts [hunk 3]
```diff
@@ -31,31 +31,17 @@ export type NotarizeOptions = NotarizeStartOptions;
...
-    );
-    if (zipResult.code !== 0) {
-      throw new Error(`Failed to zip application, exited with code: ${zipResult.code}\n\n${zipResult.output}`);
-    }
-    d('zip succeeded, attempting to upload to apple');
-
+  //zip the app unless it is dmg
+  if ( !opts.appPath.endsWith(`.dmg`) ){
+    appPath = await zipApp(dir, opts.appPath);
+  }
```
Comment (malept) [inline src/index.ts]: Maybe something along the lines of

```typescript
if ((await fs.stat(opts.appPath)).isDirectory()) {
  uploadPath = await zipApp(dir, opts.appPath);
} else if (!opts.appPath.endsWith('.dmg')) {
  throw new Error('Path does not resolve to an Electron app');
}
```

Comment (MarshallOfSound) [reply src/index.ts]: Does `isDirectory` really return true for an app, thanks apple 😆

**Comment by shiqi** [comment]
> Thanks for the contribution! A few things to note:
> 
> * The code in general needs to be standardized. I've added Prettier support in #7, so once that gets merged, you should use `prettier` to format your changes.
> * A fellow member of the Docs & Tools WG, @Kilian, recently [wrote an article](https://kilianvalkhof.com/2019/electron/notarizing-your-electron-application/) about notarization where he found that you currently shouldn't sign DMGs. So I'm not currently inclined to accept this PR until folks can actually _install_ signed+notarized DMGs.

Thanks for the review! This is actually my first TypeScript PR and I certainly appreciate your time to review this. I was doing some testing on notarization of our app (for both app and dmg) on MacOS 10.15 Beta(Catalina) and based on my test results Catalina doesn't take the un-notarized dmg very well (even the app inside is notarized), it will reject your double click by default unless you right click on it and force it to open from the right click menu. But if I upload the dmg and have it notarized/stapled then it will work fine with double click.
I also noticed that the article your referred to was written mainly target MacOS 10.14.5 (Mojave) and I took some time and applied the same tests, the behavior seems different on those two versions of MacOS for me.

**Comment by shiqi** [comment]
more tests shows that we could just use the same mechanism to zip the dmg file and upload to get it notarized, it is a little extra but I think still acceptable, so I am going to close this PR as it is not necessary.

**Comment by hugozap** [comment]
@shiqi Because zipping is not used when notarizing dmg files I believe it may cause issues notarizing the zip and not the binary dmg file.

**Pull request closed without merging**